proxy/client.go

```diff
@@ -0,0 +1,75 @@
+package proxy
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+//Client an API client to perform all operations
+type Client struct {
+	ClientAuth ClientAuth
+	httpClient *http.Client
+	GatewayURL *url.URL
+	UserAgent  string
+}
+
+//ClientAuth an interface for client authentication
+type ClientAuth interface {
+	Set(req *http.Request) error
+}
+
+//NewClient initializes a new API client
+func NewClient(auth ClientAuth, gatewayURL string, TLSInsecure bool) *Client {
+
+	gatewayURL = strings.TrimRight(gatewayURL, "/")
+	baseURL, _ := url.Parse(gatewayURL)
+	client := MakeHTTPClient(&defaultCommandTimeout, TLSInsecure)
+
+	return &Client{
+		ClientAuth: auth,
+		httpClient: &client,
+		GatewayURL: baseURL,
+	}
+}
+
+//NewRequest create a new HTTP request with authentication
+func (c *Client) NewRequest(method, path string, body io.Reader) (*http.Request, error) {
```

Comment:
Do `NewRequest` or `DoRequest` need to be public? They should only be used by the sdk methods, right?

Reply:
Yes they should be private to this package.